include/rmm/mr/device/detail/free_list.hpp

```diff
@@ -191,114 +86,91 @@ struct free_list {
   bool is_empty() const noexcept { return blocks.empty(); }
 
   /**
-   * @brief Inserts a block into the `free_list` in the correct order, coalescing it with the
-   *        preceding and following blocks if either is contiguous.
+   * @brief Removes the block indicated by `iter` from the free list.
    *
-   * @param b The block to insert.
+   * @param iter An iterator referring to the block to erase.
    */
-  void insert(block const& b)
-  {
-    if (is_empty()) {
-      insert(blocks.end(), b);
-      return;
-    }
-
-    // Find the right place (in ascending ptr order) to insert the block
-    // Can't use binary_search because it's a linked list and will be quadratic
-    auto const next =
-      std::find_if(blocks.begin(), blocks.end(), [b](block const& i) { return b < i; });
-    auto const previous = (next == blocks.begin()) ? next : std::prev(next);
+  void erase(const_iterator iter) { blocks.erase(iter); }
 
-    // Coalesce with neighboring blocks or insert the new block if it can't be coalesced
-    bool const merge_prev = previous->is_contiguous_before(b);
-    bool const merge_next = (next != blocks.end()) && b.is_contiguous_before(*next);
+  /**
+   * @brief Erase all blocks from the free_list.
+   *
+   */
+  void clear() noexcept { blocks.clear(); }
 
-    if (merge_prev && merge_next) {
-      *previous = previous->merge(b).merge(*next);
-      erase(next);
-    } else if (merge_prev) {
-      *previous = previous->merge(b);
-    } else if (merge_next) {
-      *next = b.merge(*next);
-    } else {
-      insert(next, b);  // cannot be coalesced, just insert
-    }
-  }
+  /*
+   * @brief Inserts a block into the `free_list`.
+   *
+   * @param b The block to insert.
+   */
+  // void insert(block_type const& b);
 
-  /**
-   * @brief Inserts blocks from range `[first, last)` into the free_list in their correct order,
-   *        coalescing them with their preceding and following blocks if they are contiguous.
+  /*
+   * @brief Moves blocks from other into the free_list.
    *
-   * @tparam InputIt iterator type
    * @param first The beginning of the range of blocks to insert
    * @param last The end of the range of blocks to insert.
    */
-  template <class InputIt>
-  void insert(InputIt first, InputIt last)
-  {
-    std::for_each(first, last, [this](block const& b) { this->insert(b); });
-  }
+  // void insert(free_list&& other);
 
-  /**
-   * @brief Removes the block indicated by `iter` from the free list.
+  /*
+   * @brief Returns a block from `free_list` large enough to fit `size` bytes.
    *
-   * @param iter An iterator referring to the block to erase.
+   * @param size The size in bytes of the desired block.
+   * @return block A block large enough to store `size` bytes.
    */
-  void erase(const_iterator iter) { blocks.erase(iter); }
+  // block_type get_block(size_t size);
 
+  /*
+   * @brief Print all blocks in the free_list.
+   */
+  // void print();
```

Comment:
I don't understand this diff.

Reply:
Yea, the diff makes it confusing. If you ignore the red and just look at the green, all of the lines are commented out.

Reply:
Ah, These are the functions that must be implemented by any derived class. These used to be pure virtual, but I changed to static polymorphism so only the common functionality is in this class.  Since only the derived classes are used and they are not used polymorphically, I will remove these comments.